salt/metalk8s/mine/ips.sls

```diff
@@ -0,0 +1,8 @@
+{%- from "metalk8s/map.jinja" import networks with context %}
+
+Advertise control plane network IPs in the mine:
```

Comment:
Suggest to use a mine_function directly in the pillar instead of a state that push in the mine.

Can you check with @sayf-eddine-scality he asked me yesterday (and dit it btw) to have repos ip in the mine.

Quick explaination have a custom module that return the current ip (or hostname) ONLY IF the host is a control plane so maybe check pods or directly ask api server. 
And use this custom module as a mine_function so when we do `mine.update` we advertise all REAL control plane nodes automatically

Reply:
+1 Wasn't aware that's possible :blush:

Reply:
Yeah that was something I tried to do but didn't success. I'll create a debt ticket.

Reply:
#833